lib/cylc/gui/app_gcylc.py

```diff
@@ -3039,64 +3152,53 @@ def refresh_views(self):
             if view is not None:
                 view.refresh()
 
-    def create_task_filter_box(self):
+    def create_task_filter_widgets(self):
         self.task_filter_box = gtk.VBox()
-        subbox1 = gtk.HBox(homogeneous=True)
-        subbox2 = gtk.HBox(homogeneous=True)
-        self.task_filter_box.pack_start(subbox1)
-        self.task_filter_box.pack_start(subbox2)
-        padbox = gtk.HBox()
-        padbox.pack_start(self.task_filter_box, padding=10)
-        dotm = DotMaker(self.theme, size='small')
-        cnt = 0
-
-        if self.restricted_display:
-            task_states = task_state.legal_for_restricted_monitoring
-        else:
-            task_states = task_state.legal
-
-        for st in task_states:
-            ebox = gtk.EventBox()
-            box = gtk.HBox()
-            ebox.add(box)
-            icon = dotm.get_image(st)
-            cb = gtk.CheckButton(task_state.labels[st])
-            tooltip = gtk.Tooltips()
-            tooltip.enable()
-            tooltip.set_tip(cb, "Filter by task state = %s" % st)
-
-            box.pack_start(icon, expand=False)
-            box.pack_start(cb, expand=False)
-            cnt += 1
-            if cnt > (len(task_state.legal) + 1)//2:
-                subbox2.pack_start(ebox, expand=False, fill=True)
-            else:
-                subbox1.pack_start(ebox, expand=False, fill=True)
-            cb.set_active(True)
-            cb.connect('toggled', self.check_task_filter_buttons)
+        PER_ROW = 3
+        n_states = len(self.legal_task_states) 
+        n_rows =  n_states / PER_ROW
+        if n_states % PER_ROW:
+            n_rows += 1
+        dotm = DotMaker(self.theme, size=self.dot_size)
+        for row in range(0, n_rows):
+            subbox = gtk.HBox(homogeneous=True)
+            self.task_filter_box.pack_start(subbox)
+            for i in range(0, PER_ROW):
+                ebox = gtk.EventBox()
+                box = gtk.HBox()
+                ebox.add(box)
+                try:
+                    st = self.legal_task_states[row * PER_ROW + i]
+                except:
```

Comment:
`except Exception` is better, as `except:` catches stuff like `SystemExit` (buried under https://www.python.org/dev/peps/pep-0008/#programming-recommendations)